code/modules/assembly/bang.dm

```diff
@@ -0,0 +1,55 @@
+/obj/item/assembly/bang
+	name = "BANG"
+	desc = "A small electronic device able to emit a loud sound."
+	icon_state = "igniter"
+	custom_materials = list(/datum/material/iron=500, /datum/material/glass=50)
+	drop_sound = 'sound/items/handling/component_drop.ogg'
+	pickup_sound =  'sound/items/handling/component_pickup.ogg'
+	activation_delay = 40
+	var/bang_range = 2
+
+/obj/item/assembly/bang/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] is trying to ignite [user.p_them()]self with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(get_turf(src), 'sound/weapons/flashbang.ogg', 100, TRUE, 8, 0.9)
+	return BRUTELOSS
+
+/obj/item/assembly/bang/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return
+	to_chat(user, "<span class='warning'>You short out the decibel restriction circuit.</span>")
+	bang_range = 5 // This is a big mistake.
+	obj_flags |= EMAGGED
+
+/obj/item/assembly/bang/activate()
+	if(!..())
+		return FALSE
+
+	if (obj_flags & EMAGGED)
+		playsound(get_turf(src), "explosion", 200, TRUE, 12, 0.9) // when I say loud it means LOUD (normal explosion volume is 100, this is 200)
+	else
+		playsound(get_turf(src), prob(3) ? 'sound/weapons/effects/bang.ogg' : 'sound/weapons/flashbang.ogg', 100, TRUE, 8, 0.9)
```

Comment:
```suggestion
		playsound(get_turf(src), prob(3) ? 'sound/weapons/effects/bang.ogg' : 'sound/weapons/flashbang.ogg', 200, TRUE, 8, 0.9)
```
![image](https://user-images.githubusercontent.com/34602646/127217569-1c81ccd4-058b-4360-b826-011abf8200ac.png)

Reply:
fuck you, I am turning the volume up.

Reply:
based